src/imitation/policies/trainer.py

```diff
@@ -45,6 +46,42 @@ def train(self, steps: int, **kwargs):
                 the training procedure.
         """
 
+    def set_logger(self, custom_logger: Optional[logger.HierarchicalLogger]):
```

Comment:
We used `@property` and `@property.setter` in `BaseImitationAlgorithm` so maybe stick to that convention (although, to be clear, both approaches seem fine in general)? Unless you want `logger` getter to not be exposed in public API?

Reply:
Good point, I switched to properties now